catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -211,34 +154,87 @@ public void setEnableBackupPlugin(boolean enablePlugin) {
      * @param dir absolute path for the root file system backup directory.
      */
     public void setRootBackupDir(String dir) {
-        if (StringUtils.isBlank(dir)) {
-            LOGGER.error("The root backup directory is blank.");
-            return;
+        Validate.notNull(dir);
+        File directory = new File(dir);
+        validateDirectory(directory);
+        this.rootBackupDir = directory;
+        LOGGER.trace("Set root backup directory to: {}", this.rootBackupDir.toString());
+    }
+
+    ExecutorService getExecutor() {
+        return executor;
+    }
+
+    void setExecutor(ExecutorService executor) {
+        this.executor = executor;
+    }
+
+    private synchronized void execute(Runnable task) {
+
+        executor.execute(task);
+    }
+
+    private void create(List<Metacard> metacards) {
+
+        List<String> errors = new ArrayList<>();
+        for (Metacard metacard : metacards) {
+            try {
+                createFile(metacard);
+            } catch (RuntimeException | IOException e) {
+                errors.add(metacard.getId());
+            }
         }
 
-        this.rootBackupDir = new File(dir);
-        LOGGER.debug("Set root backup directory to: {}", this.rootBackupDir.toString());
+        if (!errors.isEmpty()) {
+            LOGGER.debug(getExceptionMessage(errors, CREATE));
```

Comment:
We should log at least warnings when we fail to update the backup files, otherwise admins won't be able to know that their backup is incomplete.

Applies to the other methods below as well.